### PR TITLE
Refactored IPN/webhook handlers for subscription cancelled messages

### PR DESCRIPTION
**Pull request by dparker1005**
### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:
Previously, all of our IPN/webhook handlers were coded in vacuums causing a lot of similar code across the platform. This change is the first step in making these IPN/webhook handlers more modular so that they are easier to develop and update in the future.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: includes/gateway_request_handlers.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * @param string $gateway_environment 'live' or 'sandbox'.
+ *
+ * @return string Entry to add to IPN/webhook log.
+ */
+function pmpro_handle_subscription_cancellation_at_gateway( $subscription_transaction_id, $gateway, $gateway_environment ) {
+	// Find subscription.
+	$subscription = PMPro_Subscription::get_subscription_from_subscription_transaction_id( $subscription_transaction_id, $gateway, $gateway_environment );
+	if ( empty( $subscription ) ) {
+		// The subscription does not exist on this site. Bail.
+		return 'ERROR: Could not find this subscription to cancel (subscription_transaction_id=' . $subscription_transaction_id . ').';
```
Comment (andrewlimaza) [inline includes/gateway_request_handlers.php:20]: Whenever we return strings it's good to wrap them in i18n WP functions (unless we rely on the error text being in English always).

File: services/ipnhandler.php [hunk 2]
```diff
@@ -241,66 +241,9 @@
...
-				//send an email to the admin
-				$myemail = new PMProEmail();
-				$myemail->sendCancelAdminEmail( $user, $last_subscription_order->membership_id );
-			}
-		}
-
-		pmpro_ipnExit();
-	}
+	// Find subscription.
+	ipnlog( pmpro_handle_subscription_cancellation_at_gateway( $recurring_payment_id, 'paypalexpress', $gateway_environment ) );
```
Comment (andrewlimaza) [inline services/ipnhandler.php:245]: Might be worthwhile to rewrite `ipnlog` to be something like `pmpro_ipnlog` - had this issue before with .org saying we should prefix functions like this. Just noting it here while we refactor the IPN files.

File: includes/gateway_request_handlers.php (hunk 1 above)
Comment (dparker1005) [reply includes/gateway_request_handlers.php:20]: Discussed in Slack, logs should be in English (at least for now).

File: services/ipnhandler.php (hunk 2 above)
Comment (dparker1005) [reply services/ipnhandler.php:245]: Discussed in Slack, putting of this change for a later refactor. Could create generic logging functionality in the future.

**Review by andrewlimaza — APPROVED** [review]

**Comment by andrewlimaza** [comment]
Looks good, @ideadude.

**Pull request merged**